src/LanguageContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import * as RNLocalize from 'react-native-localize';
+import {useState, useEffect} from 'react';
+import {Language, lobot, DEFAULT_LANGUAGE} from './utils/language';
+import {useRemoteConfig} from './RemoteConfigContext';
+
+const AppLanguageProvider: React.FC = ({children}) => {
+  const currentLanguage = useLanguage();
+  const {enable_i18n} = useRemoteConfig();
+  if (!enable_i18n) {
+    return <>{children}</>;
+  }
+  return (
+    <lobot.LanguageProvider value={currentLanguage}>
+      {children}
+    </lobot.LanguageProvider>
+  );
+};
+function useLanguage() {
+  const [currentLanguage, setCurrentLanguage] = useState(DEFAULT_LANGUAGE);
+  const [locale, setLocale] = useState(preferredLocale);
+
+  const onChange = () => {
```

Comment:
Nitpick mode: I think this can be moved inside useEffect where it is used. That way we don't recreate it every time

Reply:
will do